app/src/main/java/ai/elimu/appstore/service/ProgressResponseBody.java

```diff
@@ -0,0 +1,65 @@
+package ai.elimu.appstore.service;
+
+import java.io.IOException;
+
+import okhttp3.MediaType;
+import okhttp3.ResponseBody;
+import okio.Buffer;
+import okio.BufferedSource;
+import okio.ForwardingSource;
+import okio.Okio;
+import okio.Source;
+
+/**
+ * HTTP response body which supports progress update when downloading big files
+ */
+public class ProgressResponseBody extends ResponseBody {
```

Comment:
Why is this file added if it's already included in the OkHttp library?: https://github.com/square/okhttp/blob/209c6c5589d7b0a0f10b4ce78ac6be0740a5cea8/samples/guide/src/main/java/okhttp3/recipes/Progress.java#L68

Reply:
It's not included in OkHttp library.
The file above is within sample folder.

okhttp/samples/guide/src/main/java/okhttp3/recipes/Progress.java